### ISSUE TITLE
Add continuous integration

### ISSUE THREAD

**Issue by valentina-s** [orcasound/orcaal#2]
Since different services: frontend, backend are written in different languages, maybe a first step would be to use different continuous integration providers for each component. There are already several docker containers which can be leveraged.

**Comment by Benjamintdk** [comment]
Hi @valentina-s, 

I would like to try and tackle this issue! I'm tentatively planning to do up the CI workflow using Github Actions. 

Also, would it be prudent to add a docker-compose file in the root directory to build and run all the different services at once? I can do one up as well if needed. Thanks!

**Comment by valentina-s** [comment]
Hi @Benjamintdk, thanks for willing to contribute to OrcaAL! It would be great to set up Continuous Integration with Github Actions. Having a `docker-compose` will be also very helpful. Initially we set up individual docker containers since we thought it might be easier for people to just grab and reuse separate components of the system and reuse them. But last summer we realized it can be a bit tedious starting the system so we actually discussed having a compose functionality. You can add is a separate issue. Let me know if you need more github rights, I think you can start testing in a fork. And feel free to ask any questions!

**Comment by Benjamintdk** [comment]
@valentina-s thanks for the clarifications! I have opened a new issue on implementing docker compose, and will work on implementing the Github Actions workflow first!

I have 1 further question before implementing the workflows:

I was thinking of using docker build push actions(https://github.com/docker/build-push-action) to deploy the docker images to dockerhub. This would require the use of a dockerhub account username and the corresponding password stored in the repository's secrets variables. Just wanted to clarify if there are any secrets configured for this repo, and if there is an official dockerhub account to push the built images to?

**Comment by Benjamintdk** [comment]
I've created a PR #13 for the basic Github Actions workflow! It is a little bigger than I anticipated as I did run into a few indecisions and issues while creating the basic workflow, and had to change quite a few files as well to fix the linting and resolution of paths, etc. I was also not able to pass the last step of the workflow of deploying the web app to Github pages, I suspect most likely due to lack of access permission.

Also, I left out end-to-end and unit testing in this PR, because I think the former makes more sense to do with all the containers running simultaneously (with the docker-compose), and the latter because some tests are broken and I've already made quite a few changes in the current PR. 

Do take a look and let me know if there is any further room for improvement/stuff I missed out!
